RFS/src/test/java/com/rfs/integration/EndToEndTest.java

```diff
@@ -1,67 +1,117 @@
+            var bothClustersStarted = CompletableFuture.allOf(
+                CompletableFuture.runAsync(() -> sourceCluster.start()),
+                CompletableFuture.runAsync(() -> targetCluster.start()));
+            bothClustersStarted.join();
+
+            // Setup
+            var sourceClusterOperations = new ClusterOperations(sourceCluster.getUrl());
+            var templateName = "my_template_foo";
+            sourceClusterOperations.createLegacyTemplate(templateName, "bar*");
+            var indexName = "barstool";
+            // Creates a document that uses the template
+            sourceClusterOperations.createDocument(indexName, "222", "{\"hi\":\"yay\"}");
+
+            // Take a snapshot
+            var snapshotName = "my_snap";
+            var sourceClient = new OpenSearchClient(sourceCluster.getUrl(), null, null, true);
+            var snapshotCreator = new FileSystemSnapshotCreator(snapshotName, sourceClient, SearchClusterContainer.CLUSTER_SNAPSHOT_DIR);
+            SnapshotRunner.runAndWaitForCompletion(snapshotCreator);
+            sourceCluster.copySnapshotData(localDirectory.toString());
+
+            var sourceRepo = new FileSystemRepo(localDirectory.toPath());
+            var targetClient = new OpenSearchClient(targetCluster.getUrl(), null, null, true);
+
+            var repoDataProvider = new SnapshotRepoProvider_ES_6_8(sourceRepo);
+            var metadataFactory = new GlobalMetadataFactory_ES_6_8(repoDataProvider);
+            var metadataCreator = new GlobalMetadataCreator_OS_2_11(targetClient, null, null, null);
+            var transformer = TransformFunctions.getTransformer(ClusterVersion.ES_6_8, ClusterVersion.OS_2_11, 1);
+            // Action
+            // Migrate metadata
+            new MetadataRunner(snapshotName, metadataFactory, metadataCreator, transformer).migrateMetadata();
+
+            // Validation
+            var targetClusterOperations = new ClusterOperations(targetCluster.getUrl());
+            var res = targetClusterOperations.get("/_template/" + templateName);
+            assertThat(res.getRight(), res.getLeft(), equalTo(200));
+            // Be sure that the mapping type on the template is an object
+            assertThat(res.getRight(), Matchers.containsString("mappings\":{"));
+
+            res = targetClusterOperations.get("/" + indexName);
+            assertThat("Shouldn't exist yet, body:\n" + res.getRight(), res.getLeft(), equalTo(404));
+
+            // Action
+            // Migrate indices
+            var indexMetadataFactory = new IndexMetadataFactory_ES_6_8(repoDataProvider);
+            var indexCreator = new IndexCreator_OS_2_11(targetClient);
+            new IndexRunner(snapshotName, indexMetadataFactory, indexCreator, transformer, List.of()).migrateIndices();
+            
+            res = targetClusterOperations.get("/barstool");
+            assertThat(res.getRight(), res.getLeft(), equalTo(200));
```

Comment:
The unit tests for transformations make a ton of sense to cover all the scenarios with a given transformation, but it still feels like we should try this stuff against a real source/target and see what happens.  It feels like we could construct a elaborate library of rarely-used transformations that succeed in our unit testing but fail when run "for real" otherwise.  The big thing here IMO is just exercising the transformation in a single case rather than every case (that's what the unit tests are for).

What do you think?